File: Linked_List/141.Linked_List_Cycle/Linked_List_Cycle.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: Optional[ListNode]) -> bool:
        hash_map = defaultdict(int)
        
        while(head):
            if hash_map[id(head)] != 0: return True
            hash_map[id(head)] += 1
            head = head.next
        return False

File: Linked_List/141.Linked_List_Cycle/Linked_List_Cycle.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        while(head){
            if(head->val == 487630) return true;
            head->val = 487630;
            head = head->next;
        }
        
        return false;
    }
};